libs/application/template-api-modules/src/lib/modules/templates/financial-statements-inao/financial-statements-inao.service.ts

```diff
@@ -1,22 +1,178 @@
-import { FinancialStatementsInaoClientService } from '@island.is/clients/financial-statements-inao'
+import {
+  CemeteryFinancialStatementValues,
+  FinancialStatementsInaoClientService,
+  PersonalElectionFinancialStatementValues,
+  PoliticalPartyFinancialStatementValues,
+} from '@island.is/clients/financial-statements-inao'
 import * as kennitala from 'kennitala'
-
 import { TemplateApiModuleActionProps } from '../../../types'
+import { getValueViaPath } from '@island.is/application/core'
+import {
+  mapValuesToIndividualtype,
+  mapValuesToPartytype,
+  mapValuesToCemeterytype,
+} from './mappers/mapValuesToUsertype'
+import { USERTYPE } from './types'
+
+const LESS = 'less'
 
+export const getCurrentUserType = (answers: any, externalData: any) => {
+  const fakeUserType: any = getValueViaPath(answers, 'fakeData.options')
+  const currentUserType: any = getValueViaPath(
+    externalData,
+    'getUserType.data.value',
+  )
+  return fakeUserType ? fakeUserType : currentUserType
+}
+
+export interface DataResponse {
+  success: boolean
+  message?: string
+}
 @Injectable()
 export class FinancialStatementsInaoTemplateService {
   constructor(
-    private financialStatementsService: FinancialStatementsInaoClientService,
+    private financialStatementsClientService: FinancialStatementsInaoClientService,
   ) {}
 
   async getUserType({ auth }: TemplateApiModuleActionProps) {
     const { nationalId } = auth
 
     if (kennitala.isPerson(nationalId)) {
-      return this.financialStatementsService.getClientType('Einstaklingur')
+      return this.financialStatementsClientService.getClientType(
+        'Einstaklingur',
+      )
     } else {
-      return this.financialStatementsService.getUserClientType(nationalId)
+      return this.financialStatementsClientService.getUserClientType(nationalId)
+    }
+  }
+
+  async submitApplication({ application, auth }: TemplateApiModuleActionProps) {
+    const { nationalId } = auth
+    const answers = application.answers
+    const externalData = application.externalData
+    const currentUserType = getCurrentUserType(answers, externalData)
+
+    if (currentUserType === USERTYPE.INDIVIDUAL) {
+      const values: PersonalElectionFinancialStatementValues = mapValuesToIndividualtype(
+        answers,
+      )
+
+      const electionId = getValueViaPath(
+        answers,
+        'election.selectElection',
+      ) as string
+      const clientName = getValueViaPath(answers, 'about.fullName') as string
+      const electionIncomeLimit = getValueViaPath(
+        answers,
+        'election.incomeLimit',
+      ) as string
+
+      const noValueStatement = electionIncomeLimit === LESS ? true : false
+
+      // actor is undefined until we add the delegate functionality to the frontend
+      const result: DataResponse = await this.financialStatementsClientService
+        .postFinancialStatementForPersonalElection(
+          nationalId,
+          undefined,
```

Comment:
The actor should also come from the auth object... and should be undefined if auth.actor.nationalId is undefined